src/openms/source/CHEMISTRY/TheoreticalSpectrumGenerator.cpp

```diff
@@ -473,16 +485,15 @@ namespace OpenMS
           dist = loss_ion.getIsotopeDistribution(FineIsotopePatternGenerator(max_isotope_probability_));
         }
 
-        // note: important to construct a string from char. If omitted it will perform pointer arithmetics on the "-" string literal
-        String ion_name = String(Residue::residueTypeToIonLetter(res_type)) + String(ion.size()) + "-" + loss_name + String((Size)abs(charge), '+');
-
         for (const auto& iso : dist)
         {
           p.setMZ(iso.getMZ() / (double)charge);
           p.setIntensity(intensity * rel_loss_intensity_ * iso.getIntensity());
           if (add_metainfo_)
           {
-            ion_names.push_back(ion_name);
+            ion_names.emplace_back(residue_str);
```

Comment:
this creates the same string in every iteration, right?
its not straight forward to avoid, I guess..

Reply:
Yes, because StringDataArray is supposed to have the same size as the ContainerType, I guess.

Reply:
Ah I misunderstood the question. I actually fixed that in the latest commit.